src/harness/testcases/WINNF_FT_S_IPR_testcase.py

```diff
@@ -905,3 +989,294 @@ def test_WINNF_FT_S_IPR_6(self, config_filename):
+
+    sas_test_harness_config = {
+        'sasTestHarnessName':
+            'SAS-Test-Harness-1',
+        'hostName':
+            getFqdnLocalhost(),
+        'port':
+            getUnusedPort(),
+        'serverCert':
+            getCertFilename('sas.cert'),
+        'serverKey':
+            getCertFilename('sas.key'),
+        'caCert':
+            getCertFilename('ca.cert'),
+        'fullActivityDumpRecords': [
+            generateCbsdRecords([device_a], [[grant_a]])
+        ]
+    }
+
+    domain_proxy = {
+        'registrationRequests': [device_b],
+        'grantRequests': [grant_b],
+        'conditionalRegistrationData': [conditionals_b],
+        'cert': getCertFilename('domain_proxy.cert'),
+        'key': getCertFilename('domain_proxy.key')
+    }
+
+    # Included only as an example; not used below.
+    esc_dpa = {
+        'dpaId': 'East4',
+        # This is the frequency range which will be activated (if applicable)
+        # and checked.
+        'frequencyRange': {
+            'lowFrequency': 3550000000,
+            'highFrequency': 3560000000
+        },
+        'points_builder': 'default (25, 10, 10, 10)',
+        'movelistMargin': 10
+    }
+
+    frequency_range = grant_a['operationParam']['operationFrequencyRange']
+    portal_dpa = {
+        'dpaId': 'BATH',
+        # This is the frequency range which will be checked. One loaded into the
+        # SAS, the DPA is automatically activated.
+        'frequencyRange': {
+            'lowFrequency': frequency_range['lowFrequency'],
+            'highFrequency': frequency_range['highFrequency'],
+        },
+        'points_builder':
+            'default (25, 10, 10, 10)',  # Not actually used since this is a single-point DPA.
+        'movelistMargin': 10
+    }
+
+    config = {
+        'dpaDatabaseConfig': dpa_database_config,
+        'sasTestHarnessConfigs': [sas_test_harness_config],
+        'domainProxies': [domain_proxy],
+        'portalDpa': portal_dpa,
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_IPR_7_default_config)
+  def test_WINNF_FT_S_IPR_7(self, config_filename):
+    config = loadConfig(config_filename)
+    self.assertValidConfig(
+        config, {
+            'dpaDatabaseConfig': dict,
+            'sasTestHarnessConfigs': list,
+            'domainProxies': list
+        }, {
+            'escDpa': dict,
+            'portalDpa': dict
+        })
+    self.assertValidConfig(
+        config['dpaDatabaseConfig'], {
+            'hostName': basestring,
+            'port': int,
+            'fileUrl': basestring,
+            'filePath': basestring
+        })
+    using_esc_dpa = 'escDpa' in config
+    using_portal_dpa = 'portalDpa' in config
+    self.assertTrue(
+        using_esc_dpa ^ using_portal_dpa,
+        msg=
+        'Invalid config: must use exactly one ESC-monitored DPA OR one portal-controlled DPA.'
+    )
+    if using_esc_dpa:
+      self.assertLessEqual(
```

Comment:
Shoudn't it be **assertEqual** instead of assertLessEqual here? (i.e raise an error if the low frequency is not 3540 MHz)

Reply:
I think assertLessEqual is correct since we need to permit activation of DPAs in 3550-3650 (whereas using assertEqual would restrict only to checking aggregate interference to always-active DPAs).

Reply:
You're right, thanks for the clarification!

Reply:
Thanks for checking! Does the rest of the PR look good?